core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -874,4 +874,17 @@ private static Cache<String, AuthSession> newSessionCache(Map<String, String> pr
             (RemovalListener<String, AuthSession>) (id, auth, cause) -> auth.stopRefreshing())
         .build();
   }
+
+  private Cache<TableIdentifier, FileIO> newFileIOCache() {
+    return Caffeine.newBuilder()
```

Comment:
Probably need to close this when the catalog.close() is called

Reply:
The cache itself doesn't have a close but it provides methods to perform clean up and invalidation. I've added calls to that when `close()` on the catalog is called.

Reply:
We'll at least need to shut down the executor, right?

Reply:
Yes we would need to shut down the executor. I left the `.scheduler(..)` configuration here from previous experimentation, but wanted to remove it because it's requiring some additional code for shutdown. 
It might be ok if maintenance operations run whenever the cache is being accessed. wdyt?

Reply:
Agreed. I don't think we need it.